lib/collection/src/operations/cluster_ops.rs

```diff
@@ -196,3 +204,12 @@ pub struct AbortShardTransfer {
     pub to_peer_id: PeerId,
     pub from_peer_id: PeerId,
 }
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, Validate)]
+pub struct StartResharding {
+    pub peer_id: Option<PeerId>,
+    pub shard_key: Option<ShardKey>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, Validate)]
+pub struct AbortResharding {}
```

Comment:
Don't we need `shard_id`, `peer_id` and `shard_key` fields here, or at least some of them?

Reply:
I now see we just take these from the current resharding state. At this point we only accept one of them to happen at a time.

We might add those fields if we eventually support multiple operations together.

Reply:
I did want to add explicit parameters, so that you can specify stuff explicitly to safe-guard from accidents and/or try to fix broken state (because internally you *have to* specify peer/shard/shard key to call `abort_resharding` methods)... but it was just way too annoying to handle, so I decided to add the most basic abort first, and I'll maybe extend this later.